src/pngwriter.cc

```diff
@@ -1065,19 +964,23 @@ void pngwriter::close()
    /* key is a 1-79 character description of type char* */
    char key_title[] = "Title";
    text_ptr[0].key = key_title;
-   text_ptr[0].text = texttitle_;
+   std::vector<char> texttitle(texttitle_.begin(), texttitle_.end());
+   text_ptr[0].text = &texttitle[0];
    text_ptr[0].compression = PNG_TEXT_COMPRESSION_NONE;
    char key_author[] = "Author";
    text_ptr[1].key = key_author;
-   text_ptr[1].text = textauthor_;
+   std::vector<char> textauthor(textauthor_.begin(), textauthor_.end());
+   text_ptr[1].text = &textauthor[0];
```

Comment:
nope, [c_str()](http://www.cplusplus.com/reference/string/string/c_str/) points to an *internal* array object and "may be invalidated *by further calls* to *other* member functions that *modify the object*".